service/history/replication/task_ack_manager.go

```diff
@@ -470,7 +470,7 @@ func (t *taskAckManagerImpl) generateFailoverMarkerTask(
 		TaskType:     types.ReplicationTaskType.Ptr(types.ReplicationTaskTypeFailoverMarker),
 		SourceTaskID: common.Int64Ptr(taskInfo.GetTaskID()),
 		FailoverMarkerAttributes: &types.FailoverMarkerAttributes{
-			DomainID:        common.StringPtr(taskInfo.GetDomainID()),
```

Comment:
some of these original lines are a bit facepalm-ridden :|
e.g. this one takes a nullable `*string`, converts it to a non-null `string` defaulting to `""`, then gets a pointer back to `*string`.  effectively just forgetting if the original was null or not.

well.  this is why I like this PR, I suppose.